octree/include/pcl/octree/octree_iterator.h

```diff
@@ -167,6 +167,23 @@ class OctreeIteratorBase {
     }
   }
 
+  /** \brief Preincrement operator.
+   * \note step to next octree node
+   */
+  virtual OctreeIteratorBase&
+  operator++() = 0;
```

Comment:
In clang-tidy, the following error exists:
```
/__w/pcl/pcl/octree/include/pcl/octree/octree_iterator.h:180:3: error: return type 'OctreeIteratorBase<OctreeT>' is an abstract class [clang-diagnostic-error]
  operator++(int)
```
It seems this function cannot return `OctreeIteratorBase` itself because it's an abstract class. So what other function can be made pure virtual?

Reply:
The error is for the postincrement operator (line 180). The pure virtual preincrement operator should be possible. And maybe a pure virtual postincrement operator (no implementation) is also possible? Otherwise we can only define a pure virtual preincrement operator, then we still achieve what we wanted.

Reply:
> The pure virtual preincrement operator should be possible.

There's another message on preincrement operator, not sure if it's an error:
```
/__w/pcl/pcl/octree/include/pcl/octree/octree_iterator.h:174:3: note: unimplemented pure virtual method 'operator++' in 'OctreeIteratorBase'
  operator++() = 0;
  ^
```

Reply:
After the commit `[feat] pure virtual postincrement operator]`, the errors become:
```
/__w/pcl/pcl/octree/include/pcl/octree/octree_iterator.h:180:3: error: return type 'OctreeIteratorBase<OctreeT>' is an abstract class [clang-diagnostic-error]
  operator++(int) = 0;
  ^
/__w/pcl/pcl/octree/include/pcl/octree/octree_iterator.h:174:3: note: unimplemented pure virtual method 'operator++' in 'OctreeIteratorBase'
  operator++() = 0;
  ^
/__w/pcl/pcl/octree/include/pcl/octree/octree_iterator.h:180:3: note: unimplemented pure virtual method 'operator++' in 'OctreeIteratorBase'
  operator++(int) = 0;
  ^
```

Reply:
Okay, then let's just add a pure virtual preincrement operator and no postincrement operator. And please don't simply delete the copy-constructor/copy-assignment tests, instead rewrite them for at least one of the subclasses (e.g. `OctreeDepthFirstIterator`)

Reply:
Note that there's already `TYPED_TEST (OctreeIteratorTest, CopyConstructor)` and `TYPED_TEST (OctreeIteratorTest, CopyAssignment)` for `OctreeIteratorTypes`(a.ka. `OctreeDepthFirstIterator`, `OctreeBreadthFirstIterator`, `OctreeLeafNodeDepthFirstIterator`, `OctreeFixedDepthIterator`, `OctreeLeafNodeBreadthFirstIterator`).

Reply:
Ah yes, you are right. I missed that.